### ISSUE TITLE
NaN when training

### ISSUE THREAD

**Issue by anse0113** [JingyunLiang/HCFlow#3]
When i trained HCFlow model with DF2K dataset, it showed NaN in nll loss
(I used DF2K-tr.pklv4 & DF2K-tr_X4.pklv4 from SRFlow)
![capture](https://user-images.githubusercontent.com/31559808/130190413-c0ce85d4-7cab-464a-9cd8-7c4c4e915a53.PNG)

**Comment by JingyunLiang** [comment]
I guess you directly train HCFLow++ from scratch. As indicated in Sec.4.1 (experimental setup) of the paper, we **_fine-tune_** HCFlow++ from HCFlow. Try fine-tuning from `SR_DF2K_X4_HCFlow.pth` (I have provided the model, or you can train your own model).

Tip: 
Normalizing flow models are kind of instable in training. We have tried to improve the training stability compared with SRFlow, but it still has problems using GAN loss. See a summary as follows:
HCFlow: nll loss (stable)
HCFlow+: nll loss + pixel loss (stable)
HCFlow++: nll loss + pixel loss + gan loss + perceptual loss (kind of instable, please fine-tune from HCFlow)

**Comment by anse0113** [comment]
Thanks! I'll try what you said.

**Comment by FangGet** [comment]
@JingyunLiang I trained HCFlow for CelebA, but also get NaN for nll after several iterations.
![image](https://user-images.githubusercontent.com/11433213/136721225-ad82b708-c18e-4ba5-b3bf-b262ecdd908d.png)
Would it be possible for you to give me some advice on that?
thanks.

**Comment by JingyunLiang** [comment]
Can you give more information? Do you change anything? Do you follow my training steps (see following answer) ?

> I guess you directly train HCFLow++ from scratch. As indicated in Sec.4.1 (experimental setup) of the paper, we **_fine-tune_** HCFlow++ from HCFlow. Try fine-tuning from `SR_DF2K_X4_HCFlow.pth` (I have provided the model, or you can train your own model).
> 
> Tip: Normalizing flow models are kind of instable in training. We have tried to improve the training stability compared with SRFlow, but it still has problems using GAN loss. See a summary as follows: HCFlow: nll loss (stable) HCFlow+: nll loss + pixel loss (stable) HCFlow++: nll loss + pixel loss + gan loss + perceptual loss (kind of instable, please fine-tune from HCFlow)

**Comment by ph0316** [comment]
![image](https://user-images.githubusercontent.com/56188513/191466518-86ffee6a-d9ba-4131-8e10-35f8d5db25d6.png)

I'm sorry to bother you，when I use  HCFLow fine-tuning HCFLow++, this error occurs. The number of optimizers is wrong. How can I solve it.
